src/swtpm/swtpm.c

```diff
@@ -231,11 +234,39 @@ int swtpm_main(int argc, char **argv, const char *prgname, const char *iface)
                         "Given file descriptor type is not supported.\n");
                 exit(1);
             }
-            mlp.flags |= MAIN_LOOP_FLAG_TERMINATE | MAIN_LOOP_FLAG_USE_FD;
+            mlp.flags |= MAIN_LOOP_FLAG_USE_FD |            /* Use this as client fd */
+                         MAIN_LOOP_FLAG_TERMINATE |         /* Terminate if requested */
+                         MAIN_LOOP_FLAG_KEEP_CONNECTION;    /* Keep connection till the client connection active */
```

Comment:
I am not sure if it was bug or feature to support clients just for a single request per connection, because i have no idea how this option been used.

But i felt with that approach we cannot support pure peer-to-peer clients, hence the change to keep the connection till the client alive,  this atleast shouldn't harm the existing clients which use -f option.